### ISSUE TITLE
Wrapping `SimpleMultiPlayer` under `shared_ptr` makes it crash or not work

### ISSUE THREAD

**Issue by lucaszanella** [google/oboe#1049]
Android version(s): 8.0
Android device(s): Moto Z
Oboe version: 395f3d6ac25c2b069d53451b89dff4aa96d26eb8
App name used for testing: Drum Thumper

 
**Short description**

Making the following modifications:

https://github.com/lucaszanella/oboe/commit/c090c8d5c058f3bbb1971e4e5b5411be24c0a4f6#diff-822ecb92b114e9ac7044a99fccba256f9f01e6654e418f3e82b363529f66b40b

made the DrumThumper app stop working on my phone. No sound will come off when I touch the buttons. On an older DrumThumper app (a fork from 2019 that I've been working), instead of simply not working I get a crash right on `requestStart` of AudioStream. 

I don't see a reason on why wrapping the player in a shared_ptr would give this problem:

`static SimpleMultiPlayer sDTPlayer`

to

`static std::shared_ptr<SimpleMultiPlayer> sDTPlayer = std::make_shared<SimpleMultiPlayer>();`

 I did these modifications because I want to create more than 1 drum thumper instance on my app, so I used a hashmap of `SimpleMultiPlayer` instances but I was getting a crash and so I reduced the problem to this.

I also tried to do this:

`static std::map<int, SimpleMultiPlayer> sDTPlayers`

on the older app and I got crashes in the same way

**Steps to reproduce**
Just launch
**Expected behavior**
Play sounds
**Actual behavior**
Do not play sounds, or crash on other example
**Device**
Moto Z
Please list which devices have this bug.
If device specific, and you are on Linux or a Macintosh, connect the device and please share the result for the following script. This gets properties of the device. 

```
for p in \
    ro.product.brand ro.product.manufacturer ro.product.model \
    ro.product.device ro.product.cpu.abi ro.build.description \
    ro.hardware ro.hardware.chipname ro.arch "| grep aaudio";
    do echo "$p = $(adb shell getprop $p)"; done
```

**Any additional context**

If applicable, please attach a recording of the sound.

**Comment by philburk** [comment]
I don't see why this would crash.
I'm also not sure what creating a static std::shared_ptr does for you.

If you want to prevent the objects from getting deleted then you could declare multiple static objects.

static SimpleMultiPlayer sDTPlayers[NUM_PLAYERS];

**Comment by android-usb** [comment]
Hi Lucas,

I'm not sure what making the singleton sDTPlayer a shared pointer does for you.

If I understand correctly, you want to play more sounds that the 8 that DrumThumper plays. To do this you don't need multiple players (in fact, you NEVER want multiple players), rather you want to add more signal sources in the mSampleBuffers and mSampleSources arrays.

SimpleMultipPlayer will mix as many signal sources as there are active (triggered) signal sources in those arrays.

Let me know if I can clarify further.

**Comment by lucaszanella** [comment]
@android-usb yes I just wanted to make the task loading and unloading soundpacks easier by creating a separate player for each of them.

The idea of @philburk of having  an array `static SimpleMultiPlayer sDTPlayers[NUM_PLAYERS];` works but I was very puzzled about why I'd get crashes by doing the same with a `std::map<int, SimpleMultiPlayer>`. I made it such that if no player existed at a key I'd create it. Other than that I made no modifications and still got crashes... Anyways, it's working with a static array.
